src/compiler/checker.ts

```diff
@@ -5885,22 +5888,20 @@ module ts {
         function inferTypeArguments(signature: Signature, args: Expression[], excludeArgument?: boolean[]): InferenceContext {
             var typeParameters = signature.typeParameters;
             var context = createInferenceContext(typeParameters, /*inferUnionTypes*/ false);
-            var mapper = createInferenceMapper(context);
-            // First infer from arguments that are not context sensitive
+            var inferenceMapper = createInferenceMapper(context);
+
+            // First infer from all arguments using wildcards for all context sensitive function expressions

```

Comment:
Make this comment `First infer from all arguments, but use wildcards for all context sensitive expressions.`

The "but" actually would really help to point out that this is a special case.